azure-quantum/azure/quantum/plugins/cirq/targets/ionq.py

```diff
@@ -0,0 +1,170 @@
+
+if TYPE_CHECKING:
+    from azure.quantum import Workspace
+    from azure.quantum.job import Job as AzureJob
+
+
+class _IonQClient:
+    """Thin wrapper around Workspace to support cirq_ionq.Job"""
+    def __init__(self, workspace: "Workspace"):
+        self._workspace = workspace
+    
+    @staticmethod
+    def _to_ionq_status(status: str):
+        from azure.quantum._client.models._quantum_client_enums import JobStatus
+        _STATUS_DICT = {
+            JobStatus.SUCCEEDED: 'completed',
+            JobStatus.CANCELLED: 'canceled',
+            JobStatus.FAILED: 'failed',
+            JobStatus.EXECUTING: 'running',
+            JobStatus.WAITING: 'ready'
+        }
+        return _STATUS_DICT.get(status, 'submitted')
+    
+    def _create_job_dict(self, azure_job: "AzureJob"):
+        metadata = azure_job.details.input_params.copy()
+        metadata.update(azure_job.details.metadata or {})
+
+        if azure_job.has_completed():
+            results = azure_job.get_results()
+        else:
+            results = None
+
+        job_dict = {
+            "id": azure_job.id,
+            "name": azure_job.details.name,
+            "status": self._to_ionq_status(azure_job.details.status),
+            "data": results,
+            "metadata": metadata,
+            "target": azure_job.details.target,
+            "qubits": metadata.get("qubits"),
+            "failure": azure_job.details.error_data or ""
+        }
+
+        return job_dict
+    
+    def get_job(self, job_id: str):
+        azure_job = self._workspace.get_job(job_id)
+        return self._create_job_dict(azure_job)
+
+    def cancel_job(self, job_id: str):
+        azure_job = self._workspace.get_job(job_id)
+        self._workspace.cancel_job(azure_job)
+
+    def delete_job(self, job_id: str):
+        azure_job = self._workspace.get_job(job_id)
+        self._workspace.cancel_job(azure_job)
+
+
+class IonQTarget(IonQ, CirqTarget):
+    """Base class for interfacing with an IonQ backend in Azure Quantum"""
+
+    def __init__(
+        self,
+        workspace: "Workspace",
+        name: str,
+        input_data_format: str = "ionq.circuit.v1",
+        output_data_format: str = "ionq.quantum-results.v1",
+        provider_id: str = "IonQ",
+        content_type: str = "application/json",
+        encoding: str = "",
+        **kwargs
+    ):
+        self._client = _IonQClient(workspace)
+        super().__init__(
+            workspace=workspace,
+            name=name,
+            input_data_format=input_data_format,
+            output_data_format=output_data_format,
+            provider_id=provider_id,
+            content_type=content_type,
+            encoding=encoding,
+            **kwargs
+        )
+
+    @staticmethod
+    def _translate_cirq_circuit(circuit) -> Dict[str, Any]:
+        """Translate Cirq circuit to IonQ JSON. If dependencies \
+    are not installed, throw error with installation instructions."""
+        try:
```

Comment:
is the `try` necessary, we already that at the top of the module

Reply:
Yes I agree it's redundant and can be removed. Thanks for catching that.